src/Symfony/Component/Finder/Iterator/FilterIterator.php

```diff
@@ -18,9 +18,24 @@
  *
  * @author Alex Bogomazov
  */
-abstract class FilterIterator extends \FilterIterator
+abstract class FilterIterator extends \FilterIterator implements \RecursiveIterator
```

Comment:
It's not broken at all, it's just a new behavior that didn't exist before and can be changed easily by wrapping the injected recursive iterator in a RecursiveIteratorIterator. I see no issue here. To get the behavior you need is really opt-in: inject a recursive iterator, and use this one as a recursive iterator also. I wouldn't say anyone will do this by mistake.